src/dev/bazel/jsts_transpiler.bzl

```diff
@@ -2,7 +2,7 @@
 
 load("@npm//@babel/cli:index.bzl", _babel = "babel")
 
-def jsts_transpiler(name, srcs, build_pkg_name, web = False, root_input_dir = "src", additional_args = ["--quiet"], **kwargs):
+def jsts_transpiler(name, srcs, build_pkg_name, web = False, root_input_dir = "", additional_args = ["--quiet"], **kwargs):
```

Comment:
I'd like to understand why we are removing the `src` dir. TS projects usually include all the TS src files under `src`. AFAIK, It's a well-established convention.

I may understand we may want to remove the limitation of having `src` for non-TS packages. However, this is the JS-TS transpiler, and most of our packages use TS. Wouldn't it be best if non-TS packages explicitly report `root_input_dir = ""` instead of the other way around?

Reply:
Thanks for the feedback @afharo! As part of the IDM project we want to make packages as closer as we can to the rest of the code in the repo at the same time we also want to make it easier to create packages. The decision to remove the restriction we had previously on the `src` folder came up in a discussion between me and @spalger after getting feedback from other folks on the Core and the shared-ux team. We feel like by not having that restriction in place we will be making the life easier for people creating packages. Having that said, if you feel like you still want to contain your code inside a given folder lets say, `src` or `impl` you can still do it as long as your `index{js,ts}` file lives at the root of the repo.

Reply:
I understand the benefits. Thanks for explaining :elasticheart:

I'm more concerned about how the IDEs will behave in that scenario. I can see Bazel won't really struggle because the out dirs are elsewhere. But IDEs relying on tsconfig.json... how would they work?

If it's controlled and no issues have been found in your testing. I'm not opposed at all to this change (I'm genuinely happy about it indeed)

Reply:
I trust you're going to test it thoroughly :)

All good on my end then 👍